js/ui/themes.js

```diff
@@ -301,24 +307,38 @@ function detachCssClasses(element) {
     $(element).removeClass(themeClasses);
 }
 
-init({
-    _autoInit: true,
-    _forceTimeout: true
-});
-
 function themeReady(callback) {
     themeReadyCallback.add(callback);
 }
 
-ready(function() {
+var readyDeferred = new Deferred();
+var initDeferred = new Deferred();
+
+readyDeferred.done(function() {
+    init({
+        _autoInit: true,
+        _forceTimeout: true
+    });
+
     if($(DX_LINK_SELECTOR, context).length) {
         throw errors.Error("E0022");
     }
+    initDeferred.resolve();
 });
 
+if(windowUtils.hasWindow()) {
+    readyDeferred.resolve();
+} else {
+    ready(function() {
+        readyDeferred.resolve();
+    });
+}
+
 viewPortChanged.add(function(viewPort, prevViewPort) {
-    detachCssClasses(prevViewPort);
-    attachCssClasses(viewPort);
+    initDeferred.done(function() {
+        detachCssClasses(prevViewPort);
+        attachCssClasses(viewPort);
```

Comment:
In this case, detachCssClasses and attachCssClasses methods will be called only once. In the previous version, it could be called several times. I think that this change can cause problems